lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs

```diff
@@ -102,8 +102,28 @@ impl Client {
     /// It updates the local clipboard cache and returns the encoded message
     /// that should be sent to the RDP server.
     pub fn update_clipboard(&mut self, data: Vec<u8>) -> RdpResult<Vec<u8>> {
+        const CR: u8 = 0x13;
+        const LF: u8 = 0x10;
+
+        // convert LF to CRLF, as required by CF_OEMTEXT
+        let len_orig = data.len();
+        let mut converted = Vec::with_capacity(len_orig);
+        for i in 0..len_orig {
+            match data[i] {
+                LF if (i != 0) && data[i - 1] != CR => {
+                    converted.push(CR);
+                    converted.push(LF);
+                }
+                _ => converted.push(data[i]),
+            }
+        }
+        // Windows requires a null terminator, so add one if necessary
+        if converted[converted.len() - 1] != 0x00 {
```

Comment:
I assume this should be guarded for zero length data.

```suggestion
        if !converted.is_empty() && converted[converted.len() - 1] != 0x00
```